src/actions/EqualValueEmbeddedConstraintAction.C

```diff
@@ -0,0 +1,89 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "EqualValueEmbeddedConstraintAction.h"
+#include "Conversion.h"
+#include "FEProblem.h"
+#include "Factory.h"
+#include "MooseMesh.h"
+
+#include "libmesh/string_to_enum.h"
+// #include <algorithm>
+
+registerMooseAction("BlackBearApp", EqualValueEmbeddedConstraintAction, "add_constraint");
+
+InputParameters
+EqualValueEmbeddedConstraintAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Sets up constraint on primary and secondary variables on "
+                             "primary and secondary block");
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "primary_block", "The list of ids of the primary block (subdomain)");
```

Comment:
To make this more consistent, I'd suggest keeping `primary`, `secondary` rather than introducing the new `primary_block` and `secondary_block` parameters.